packages/zoe/docs/AttackGuide.md

```diff
@@ -0,0 +1,154 @@
+# An Attacker's guide to Zoe
+
+This is an incomplete list of potential weak points that an attacker might want to focus
+on when looking for ways to violate Zoe's integrity. It's here to help defenders, as
+"attacker's mindset" is a good way to focus attention for the defender. The list should
+correspond pretty closely to the set of assurances that Zoe aims to support.
+
+The main focus of most threats would be a breach of one of Zoe's core invariants
+
+ * reallocations ordered by contracts must not violate
+   * offer safety
+   * exit rules, or
+   * rights conservation
+ * invitations are needed in order to exercise offers
+ * each instance of a contract is isolated from others and can only communicate through
+    capabilities that have been explicitly shared
+
+
+## Reallocation
+
+The current approach (incrementBy/decrementBy and the fact that all seats must be included
+in realloc) has led to a few bugs. It's probably worth looking for other cases that
+create new stagings or presume there are none outstanding.
+
+Notice that `reallocateForZcfMint()` is allowed to violate rights conservation since its
+minting new value. Can it be confused into including other reallocations with the new
+issuance?
+
+reallocate in `zcfSeat.js` is not atomic. It starts by calculating the reallocations for
+every seat, then updates them one at a time (replaceAllocations in
+`startInstance.js`). There's a try-catch around the loop, so an outright failure to
+reallocate any seat would cause the vat to be terminated. Are there any failures of an
+individual reallocation that would violate the consistency rules for the overall
+reallocation?
+
+
+## Asynchrony
+
+Zoe has to manage some complex interactions to achieve its ends, so finding gaps in the
+handling of asynchrony between Zoe and ZCF is a possible avenue of attack.
+
+## Issuers
+
+We intend to make issuers less interdependent. Can you break zoe's escrow with a misbehaving issuer?
+
+## feeMintAccess
+
+When zoe starts up, the creator gets feeMintAccess and initialFeeFunds. Those are powerful
+caps. Can they be extracted from bootstrap?
+
+When starting zoe, the caller can supply a feeIssuerConfig. If they could specify a higher
+initialFunds than needed, and squirrel away the excess, no one else would know.
+
+## Non-responsive contracts
+
+Someone should demonstrate a subtle way of causing a non-responsive contract that would
+surprise clients. If it's too obvious, no one will use the contract.
```

Comment:
```suggestion
surprise clients. If it's too obvious, no one will use the contract.
```
"If it's too obvious" is a strange way to put it. Makes it sound like a virtue for the flaw to not be obvious. How about
```suggestion
surprise clients. If the contract could be made non-responsive, no one should it.
```
Or, just omit that last sentence. I'm not sure it adds anything.

Reply:
I dropped it.